colossalai/auto_parallel/tensor_shard/sharding_strategy.py

```diff
@@ -84,6 +85,23 @@ class MemoryCost:
     buffer: int = 0
 
 
+class CommType(Enum):
+    """
+    CommType describes the sequential order of a communication action and a computation action.
+    """
+    BEFORE = 0
+    AFTER = 1
+    HOOK = 2
+    IMPLICIT = 3
+
+
+@dataclass
+class CommAction:
+    comm_spec: CommSpec = None
+    comm_type: CommType = None
+    arg_index: int = -1
```

Comment:
What is `arg_index`?

Reply:
I will add a comment to explain it

Reply:
Why `arg_index` is needed? If the args of the node are changed, we should update the sharding strategy data accordingly instead of bookkeeping its order.